Eatery/Views/Eateries/ScrollableTextTabBarControl.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  ScrollableTextTabBarController.swift
+//  Eatery
+//
+//  Created by Sergio Diaz on 11/11/20.
+//  Copyright © 2020 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class ScrollableTextTabBarControl: TabBar {
+
+    let padding: CGFloat!
+    let underlineOffset: CGFloat = 6
+
+    init(sections: [String], padding: CGFloat = 0) {
+        self.padding = padding
+        super.init(sections: sections)
+
+        for section in sections {
+            let tabButton = UIButton()
+            tabButton.setTitle(section, for: .normal)
+            tabButton.setTitleColor(.gray, for: .normal)
+            tabButton.titleLabel?.font = UIFont.systemFont(ofSize: 20.0, weight: .semibold)
+            tabButton.contentEdgeInsets = UIEdgeInsets(top: 0, left: 12, bottom: 0, right: 12)
+            tabButton.addTarget(self, action: #selector(tabButtonPressed(_:)), for: .touchUpInside)
+            tabButton.frame.size.height = frame.height
+            tabButton.clipsToBounds = true
+            tabButton.sizeToFit()
+            tabButtons.append(tabButton)
+        }
+
+        scrollView = UIScrollView()
+        scrollView.isScrollEnabled = true
+        scrollView.alwaysBounceHorizontal = true
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+
+        scrollView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+
+        stackView = UIStackView(arrangedSubviews: tabButtons)
+        stackView.spacing = 0
+        stackView.axis = .horizontal
+        stackView.backgroundColor = .white
+        scrollView.addSubview(stackView)
+
+        stackView.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(padding)
+            make.top.bottom.trailing.equalToSuperview()
+        }
+
+        underlineView = UIView()
+        underlineView.backgroundColor = .eateryBlue
+        scrollView.addSubview(underlineView)
+
+        select(at: 0)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func select(at index: Int) {
+        // Base case to protect against accidental backend errors
+        if tabButtons.count == 0 {
```

Comment:
Slight nit but I think this is the more swifty way to do it: makes clear that you expect `tabButtons` to be non-empty

```suggestion
        guard !tabButtons.isEmpty else {
```